apis/core/v1beta1/openstackversion_types.go

```diff
@@ -288,3 +292,92 @@ func GetOpenStackVersions(namespace string, k8sClient client.Client) (*OpenStack
 
 	return versionList, nil
 }
+
+// isContainerTemplateEmpty checks if all fields in a ContainerTemplate are nil
+func isContainerTemplateEmpty(ct ContainerTemplate) bool {
+	v := reflect.ValueOf(ct)
+	numFields := v.NumField()
+	for i := 0; i < numFields; i++ {
+		field := v.Field(i)
+		// Check if field is a pointer and not nil
+		if field.Kind() == reflect.Ptr && !field.IsNil() {
+			return false
+		}
+	}
+	return true
+}
+
+// customContainerImagesModified compares two CustomContainerImages and returns true if they are different
+func customContainerImagesAllModified(a, b CustomContainerImages) bool {
+	if !containerTemplateEqual(a.ContainerTemplate, b.ContainerTemplate) {
+		return true
+	}
+
+	if !stringMapEqual(a.CinderVolumeImages, b.CinderVolumeImages) {
+		return true
+	}
+
+	if !stringMapEqual(a.ManilaShareImages, b.ManilaShareImages) {
+		return true
+	}
+
+	// If all fields are equal, return false (not modified)
+	return false
+}
+
+// containerTemplateEqual compares two ContainerTemplate structs for equality using reflection
+func containerTemplateEqual(a, b ContainerTemplate) bool {
+	va := reflect.ValueOf(a)
+	vb := reflect.ValueOf(b)
+
+	numFields := va.NumField()
+	for i := 0; i < numFields; i++ {
+		fieldA := va.Field(i)
+		fieldB := vb.Field(i)
+
+		// Both fields should be *string type
+		if fieldA.Kind() != reflect.Ptr || fieldB.Kind() != reflect.Ptr {
+			continue
+		}
+
+		// Compare using stringPtrEqual logic
```

Comment:
Did you mean to call `stringPtrEqual()`, or are you just saying that you're using the func as inspiration for the particular `if` statements that follow?